packages/babel-plugin/__tests__/stylex-transform-define-consts-test.js

```
@@ -0,0 +1,354 @@
+        stylexPlugin,
+        {
+          runtimeInjection: true,
+          unstable_moduleResolution: { type: 'haste' },
+          ...defaultOpts,
+          ...opts,
+        },
+      ],
+    ],
+  });
+}
+
+describe('@stylexjs/babel-plugin', () => {
+  describe('[transform] stylex.defineConsts()', () => {
+    test('transforms constants object', () => {
+      const { code, metadata } = transform(`
+        import stylex from 'stylex';
+        export const breakpoints = stylex.defineConsts({
+          sm: '(min-width: 768px)',
+          md: '(min-width: 1024px)',
+          lg: '(min-width: 1280px)',
+        });
+      `);
+
+      expect(code).toMatchInlineSnapshot(`
+        "import stylex from 'stylex';
+        export const breakpoints = {
+          sm: "(min-width: 768px)",
+          md: "(min-width: 1024px)",
+          lg: "(min-width: 1280px)"
+        };"
+      `);
+
+      expect(metadata.stylex).toMatchInlineSnapshot(`
+        [
+          [
+            "x1izlsax",
+            {
+              "constKey": "sm",
+              "constVal": "(min-width: 768px)",
+              "ltr": "",
+              "rtl": null,
+            },
+            0,
+          ],
+          [
+            "xe5hjsi",
+            {
+              "constKey": "md",
+              "constVal": "(min-width: 1024px)",
+              "ltr": "",
+              "rtl": null,
+            },
+            0,
+          ],
+          [
+            "xmbwnbr",
+            {
+              "constKey": "lg",
+              "constVal": "(min-width: 1280px)",
+              "ltr": "",
+              "rtl": null,
+            },
+            0,
+          ],
+        ]
+      `);
+    });
+
+    test('transforms literal constants object', () => {
+      const { code, metadata } = transform(`
+        import stylex from 'stylex';
+        export const spacing = stylex.defineConsts({
+          '--small': '8px',
+          '--medium': '16px',
+          '--large': '24px',
```

This should throw an error. Disallow usage of keys that start with `--` to be used as consts.

Using `--` lets you take control of the actual strings and this could be dangerous when it comes to consts as need to guarantee that they are replaced at compile time.

---

I'm not building the key the same way as in `defineVars` (ie. key is hashed regardless of if it starts with `--` but agree it's better for clarity